src/controllers/api.js

```diff
@@ -106,4 +106,43 @@ module.exports = {
       datePosted: job.datePosted,
     });
   },
+
+  /**
+   * Fetches a single job from the system and returns it. If the job does not
+   * exist, a 404 is returned. If the requested jobId is not a number, a 400
+   * is returned.
+   *
+   * @param {JBRequest} req
+   * @param {JBResponse} res
+   */
+  async updateJob(req, res) {
+    const jobId = parseInt(req.params.jobId);
+    const jobPayload = req.body;
+
+    if (!jobId || (jobPayload.id && jobPayload.id !== jobId))
+      return res.sendStatus(400);
+    /*if (jobPayload.id && jobPayload.id !== jobId) {
+      return res.sendStatus(406);
+    }*/
+
+    const job = await req.ModelFactory.job.updateJob(
+      req.ModelFactory,
+      req.DaoFactory,
+      jobId,
+      jobPayload
+    );
```

Comment:
Instead of using a static method, this would be better implemented as an instance method.

So, fetch the job by job id, then call an update function on the instance of job model.